CONTRIBUTING.md

```diff
@@ -12,20 +12,87 @@ As a developer:
+Quoting the most relevant parts below (and thanks to the Rust maintainers for this).
+
+> Pull requests are the primary mechanism we use to change Rust. GitHub itself has some great documentation on using the Pull Request feature. We use the "fork and pull" model described here, where contributors push changes to their personal fork and create pull requests to bring those changes into the source repository.
+>
+> Please make pull requests against the master branch.
+>
+> Rust follows a no merge policy, meaning, when you encounter merge conflicts you are expected to always rebase instead of merge. E.g. always use rebase when bringing the latest changes from the master branch to your feature branch. Also, please make sure that fixup commits are squashed into other related commits with meaningful commit messages.
+>
+> GitHub allows closing issues using keywords. This feature should be used to keep the issue tracker tidy.
+
+Source: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#pull-requests
+
+Further reading:
+- https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models
+- https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests
 
-To protect the Hyperledger Cactus source code, GitHub pull requests are accepted from forked repositories only. There are also quality standards identified and documented here that will be enhanced over time.
+
+### PR Checklist - Contributor/Developer
 
 1. Fork [hyperledger/cactus](https://github.com/hyperledger/cactus) via Github UI
-1. Clone the fork to your local machine
-1. Complete the desired changes and where possible test locally (more detail to come here)
-1. Commit your changes
-    1. Make sure you sign your commit using `git commit -s` for more information see [here](https://gist.github.com/tkuhrt/10211ae0a26a91a8c030d00344f7d11b)
-    1. Make sure your commit message follows [Conventional Commits syntax](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification); this aids in release notes generation
-1. Push your changes to your master
-1. Initiate a pull request from your fork to the base repository
-1. Await CI, DCO & linting quality checks, as well as any feedback from reviewers
+2. Clone the fork to your local machine
+3. Complete the desired changes and where possible test locally
+   1. You can run the full CI suite on Mac/Linux/WSL by running the script at `./tools/ci.sh`
+   2. If you do not have your environment set up for running bash scripts, do not worry, all pull requests will automatically have the same script executed for it when opened. The only downside is the slower feedback loop.
+4. Commit your changes
+    1. Make sure you sign your commit using `git commit -S` for more information see [here](https://gist.github.com/tkuhrt/10211ae0a26a91a8c030d00344f7d11b)
```

Comment:
First of all @petermetz , thanks a lot for this guide. This is definitely useful, and it will greatly increase the productivity of all the team. 

Some feedback: I would swap the order of 4. i. and 4. ii., as there can be people going through the guide when they are committing their code.